storm-core/src/jvm/backtype/storm/scheduler/SupervisorDetails.java

```diff
@@ -119,10 +119,18 @@ private Double getTotalResource(String type) {
     }
 
     public Double getTotalMemory() {
-        return getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        Double totalMemory = getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        if (totalMemory == null) {
+            throw new IllegalStateException("default value for supervisor.memory.capacity.mb is not set!");
+        }
+        return totalMemory;
     }
 
     public Double getTotalCPU() {
-        return getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        Double totalCPU = getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        if (totalCPU == null) {
+            throw new IllegalStateException("default value for supervisor.cpu.capacity is not set!");
+        }
+        return totalCPU;
```

Comment:
> Do we want to throw exceptions here? It is unclear to me that this means the default was not set.
> 
> If we need a default value, then we should add @NotNull to the validators for these configs. Then we will see the error earlier.

Please disregard my first comment. This config does not need to be specified, and defaults.yaml is not validated like storm.conf.


When we log, we should use the String defined in Config instead of the literals.